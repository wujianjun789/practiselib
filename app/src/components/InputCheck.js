import React,{Component} from 'react';
export default class InputCheck extends Component{
    constructor(props){
        super(props)
        this.state = {
            style: { visibility: 'hidden' },
        }
        this.onFocus = this.onFocus.bind(this);
        this.onBlur = this.onBlur.bind(this);
    }

    onFocus(){
        this.setState({style:{visibility: 'hidden'}})
    }

    onBlur(event){
        this.props.onBlur && this.props.onBlur(event.target.id);
        this.setState({style:{visibility: 'visible'}});
    }

    onChange(event){
        this.props.onChange && this.props.onChange(event.target.id,event.target.value);
    }

    render(){
        const {type = 'text',className = 'userName',label = '用户名',checked = 'fail',reminder = '',placeholder = '请输入用户名',value=''} = this.props;
        const style = this.state.style?this.state.style:{ visibility: 'hidden' };

        return <div className={"inputCheck "+className}>
                <label className = 'col-sm-2 control-label'>{label}:</label>
                <div className = {`has-feedback col-sm-4 ${checked=='success'?'has-success':checked=='fail'?'has-error':''}`}>
                    <input type={type} id = {className} className="form-control" placeholder={placeholder} value={value} onFocus={this.onFocus} onBlur={this.onBlur} onChange={this.onChange}/>
                    <span className={`glyphicon ${checked=='success'?'glyphicon-ok':checked=='fail'?'glyphicon-remove':''} form-control-feedback`} aria-hidden="true"></span>
                    <span className="reminder" style={style}>{reminder}</span>
                </div>
            </div>
        
    }
}
