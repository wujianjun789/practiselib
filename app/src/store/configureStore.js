import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../common/reducer';
import thunk from 'redux-thunk';


export default function configureStore(initialState) {

    const composeEnhancers =
        process.env.NODE_ENV !== 'production' &&
            typeof window === 'object' &&
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
                // Specify here name, actionsBlacklist, actionsCreators and other options
            }) : compose;

    const enhancer = composeEnhancers(
        applyMiddleware(thunk),
        // other store enhancers if any
    );
    const store = createStore(rootReducer, initialState, enhancer);
    return store;

}
