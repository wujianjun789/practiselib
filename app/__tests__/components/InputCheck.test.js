import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl-redux';
import configureStore from '../../src/store/configureStore';

import InputCheck from '../../src/components/InputCheck';
import Immutable from 'immutable'

const store = configureStore();
describe('<InputCheck />',()=>{
    const data = {
        id:'username',
        label:'用户名',
        value:'admin',
        checked:'fail',
        reminder:'用户名已存在'
    }
    it('render normal',()=>{
        const inputCheck = shallow(<Provider store={store}>
                <IntlProvider>
                    <InputCheck label={data.label} id={data.id} value= {data.value} checked={data.checked} reminder={data.reminder}/>
                </IntlProvider>
            </Provider>);

        const container = inputCheck.find('.inputCheck');
        expect(container.length).toBe(1);

        const label = inputCheck.find('label');
        expect(label.length).toBe(1);
        // expect(label.text()).toBe(`${data.label}:`);
        
        const content = inputCheck.find('.has-feedback.has-error');
        expect(content.length).toBe(1);

        const input = inputCheck.find('input');
        expect(input.length).toBe(1);
        expect(input.prop('value')).toBe(data.value);

        const sign = inputCheck.find(`span.glyphicon.glyphicon-remove.form-control-feedback`);
        expect(sign.length).toBe(1);

        const reminder = inputCheck.find('.reminder');
        expect(reminder.length).toBe(1);
        expect(reminder.text()).toBe(data.reminder)
    })

    it('onFocus', () => {
        const onFocus = jest.fn();
        const inputCheck = shallow(<Provider store={store}>
                <IntlProvider>
                    <InputCheck label={data.label} id={data.id} onFocus={onFocus}/>
                </IntlProvider>
            </Provider>);
        let input = inputCheck.find('input');
        input.simulate('focus',{target:{id:data.id}});
        expect(onFocus).toHaveBeenCalledTimes(1);
    })

    it('onBlur', () => {
        const onBlur = jest.fn();
        const inputCheck = shallow(<Provider store={store}>
            <IntlProvider>
                <InputCheck label={data.label} id={data.id} onFocus={onFocus}/>
            </IntlProvider>
        </Provider>);
        let input = inputCheck.find('input');
        input.simulate('blur',{target:{id:data.id}});
        expect(onBlur).toHaveBeenCalledTimes(1);
    })

    it('onChange', () => {
        const onChange = jest.fn();
        const inputCheck = shallow(<Provider store={store}>
            <IntlProvider>
                <InputCheck label={data.label} id={data.id} onFocus={onFocus}/>
            </IntlProvider>
        </Provider>);
        let input = inputCheck.find('input');
        input.simulate('change',{target:{value:'a'}});
        expect(onChange).toHaveBeenCalledTimes(1);
    })

    it('snapshot', () => {
        const inputCheck = renderer.create(<Provider store={store}>
                <IntlProvider>
                    <InputCheck label={data.label} id={data.id} value= {data.value} checked={data.checked} reminder={data.reminder}/>
                </IntlProvider>
            </Provider>);
        let tree = inputCheck.toJSON();
        expect(tree).toMatchSnapshot();
    });
})