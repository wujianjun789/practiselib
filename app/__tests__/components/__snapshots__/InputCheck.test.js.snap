exports[`<InputCheck /> snapshot 1`] = `
<div
  className="inputCheck ">
  <label
    className="col-sm-4 control-label">
    用户名
    :
  </label>
  <div
    className="has-feedback col-sm-8 has-error">
    <input
      className="form-control"
      disabled={false}
      id="username"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder=""
      type="text"
      value="admin" />
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-remove form-control-feedback" />
    <span
      className="reminder">
      用户名已存在
    </span>
  </div>
</div>
`;
